docs/coding/coding_requirements.md

```diff
@@ -0,0 +1,866 @@
+   required, even if the object in question may never use it!
+
+Remember: although this tradeoff makes sense in many cases, it doesn't cover
+them all. Think carefully about your addition before deciding if you need to use
+it.
+
+## Prefer `Initialize()` over `New()` for atoms
+
+Our game controller is pretty good at handling long operations and lag, but it
+can't control what happens when the map is loaded, which calls `New()` for all
+atoms on the map. If you're creating a new atom, use the `Initialize()` proc to
+do what you would normally do in `New()`. This cuts down on the number of proc
+calls needed when the world is loaded.
+
+While we normally encourage (and in some cases, even require) bringing out of
+date code up to date when you make unrelated changes near the out of date code,
+that is not the case for `New()` -> `Initialize()` conversions. These systems
+are generally more dependent on parent and children procs, so unrelated random
+conversions of existing things can cause bugs that take months to figure out.
+
+## No implicit `var/`
+
+When you declare a parameter in a proc, the `var/` is implicit. Do not include
+any implicit `var/` when declaring a variable.
+
+```dm
+//Bad
+/obj/item/proc1(var/mob/input1, var/input2)
+    code
+
+//Good
+/obj/item/proc1(mob/input1, input2)
+    code
+```
+
+## No magic numbers or strings
+
+This means stuff like having a "mode" variable for an object set to "1" or "2"
+with no clear indicator of what that means. Make these #defines with a name that
+more clearly states what it's for. For instance:
+
+```dm
+//Bad
+/datum/proc/do_the_thing(thing_to_do)
+    switch(thing_to_do)
+        if(1)
+            do_stuff()
+        if(2)
+            do_other_stuff()
+```
+
+There's no indication of what "1" and "2" mean! Instead, you should do something
+like this:
+
+```dm
+//Good
+#define DO_THE_THING_REALLY_HARD 1
+#define DO_THE_THING_EFFICIENTLY 2
+
+/datum/proc/do_the_thing(thing_to_do)
+    switch(thing_to_do)
+        if(DO_THE_THING_REALLY_HARD)
+            do_stuff()
+        if(DO_THE_THING_EFFICIENTLY)
+            do_other_stuff()
+```
+
+This is clearer and enhances readability of your code! Get used to doing it!
+
+## Control statements
+
+- All control statements comparing a variable to a number should use the formula
+  of `thing` `operator` `number`, not the reverse (e.g. `if(count <= 10)` not
+  `if(10 >= count)`)
+- All control statements must be spaced as `if()`, with the brackets touching
+  the keyword.
+- All control statements must not contain code on the same line as the
+  statement.
+
+```dm
+//Bad
+if(x) return
+
+//Good
+if(x)
+	return
+```
+
+## Player Output
+
+Due to the use of "TGchat", Paradise requires a special syntax for outputting
+text messages to players. Instead of `mob << "message"`, you must use
+`to_chat(mob, "message")`. Failure to do so will lead to your code not working.
+
+## Use early returns
```

Comment:
We ideally prefer the term guard clause as it covers situations where return isn't used, such as continue or break